internal/lang/funcs/encoding_test.go

```diff
@@ -243,6 +243,54 @@ func TestURLEncode(t *testing.T) {
 	}
 }
 
+func TestURLDecode(t *testing.T) {
+	tests := []struct {
+		String cty.Value
+		Want   cty.Value
+		Err    bool
+	}{
+		{
+			cty.StringVal("abc123-_"),
+			cty.StringVal("abc123-_"),
+			false,
+		},
+		{
+			cty.StringVal("foo%3Abar%40localhost%3Ffoo%3Dbar%26bar%3Dbaz"),
+			cty.StringVal("foo:bar@localhost?foo=bar&bar=baz"),
+			false,
+		},
+		{
+			cty.StringVal("mailto%3Aemail%3Fsubject%3Dthis%2Bis%2Bmy%2Bsubject"),
+			cty.StringVal("mailto:email?subject=this+is+my+subject"),
+			false,
+		},
+		{
+			cty.StringVal("foo/bar"),
+			cty.StringVal("foo%2Fbar"),
+			false,
+		},
+	}
```

Comment:
I think we have to be careful here when it comes to edge cases. We cant just take the tests from urlencode and assume that the inverse covers everything.

Would you also mind adding tests for the following

- Incomplete percent encodings (ie: just `%` or `%2` and not `%20`
- Invalid percent encodings, (ie: `%GG`)
- Unicode character support (ensure we only support utf8?)
- Null bytes ie %00 should not be allowed


It may also be cool to write some tests that test the full circle of UrlEncode + UrlDecode

Reply:
Done, and I appreciate the suggestion!